### DISCUSSION TITLE
Is Feast helpful in case of the pictures and video as a features?

### DISCUSSION THREAD

**Discussion by greg-kan** [feast-dev/feast#4318] [General]
I am new to a Feature Store topic and trying to understand whether the Feast (and a Feature Store in general) is helpful if we have the pictures, video and text data as a features to teach our models?

I apologize if the question is inappropriate.

As I could understand all the feature prepairing in that case takes place inside the model (Convolution, Transformer or RNN).

I saw an article somewhere: "How to securely store an image dataset in your Feature Store" and I am misled by this fact.

Is there a kind of Feature Ftore that can ingest pictures?

**Comment by tokoko** [comment]
You can always store pictures, video or text as features of type BYTES. How useful that would turn out to be depends on what you're going to do with the features. Feature Stores in general are very useful when you have sets of features associated with some entities (think clients or users) and you often need to retrieve these features for specific entities to use during either training or inference. If that sounds like your use case, then yes. If you're only looking to train/infer models using just a single "feature" (picture, video, text, etc...) then usefulness of feature stores for you will be very limited.

A practical example would be if you're trying to classify emails into spam/ham:
- If you're trying to train a model only on email content and labels, then you can probably live without feature stores.
- If you're looking to enrich the set of features passed to the model with other precomputed features based on sender/recipient (for example IP or features calculated from previous email history), then using a feature store becomes more important.

**Reply by franciscojavierarceo** [reply]
I would generally add metadata as features to an image or video and those features would be useful and complimentary to the raw data.
